### ISSUE TITLE
QueryLog shows domain blocked when it's not

### ISSUE THREAD

**Issue by ExternalDatabase** [AdguardTeam/AdGuardHome#2686]
* **Version of AdGuard Home server:**
  * v0.104.3
* **How did you install AdGuard Home:**
  * Docker
* **How did you setup DNS configuration:**
  * IoT
* **If it's a router or IoT, please write device model:**
  * Raspberry Pi 3 Model B
* **CPU architecture:**
  * ARM
* **Operating system and version:**
  * Raspbian 10 buster

### Expected Behavior
DNS request for A record should return 0.0.0.0 for domain listed in blocklist.

### Actual Behavior
All devices connected to my network see AdGuardHome [AGH] as main (and only) DNS resolver.
AGH is seperate IoT (Raspberry Pi with Docker).
Upstream DNS server for AGH is NextDNS via DNS-over-TLS (DoT).

When device is asking AGH for A record i.e. for "target.aws.amazon.com" I can see:
in AGH QueryLog: target.aws.amazon.com - Blocked by CNAME or IP - The Block List Project - Abuse

`Response details
Status
Blocked by CNAME or IP
DNS server
tls://XXXXXXXXXX.dns.nextdns.io:853
Elapsed
47 ms
Response code
NOERROR
Filter
The Block List Project - Abuse
Rule
0.0.0.0
Response
A: 0.0.0.0 (ttl=10)
Original response
A: 0.0.0.0 (ttl=1800) `
AND AT THE SAME TIME that request was send to NextDNS, where it has been blocked.

So, how AGH can inform that it blocked request when it was passed to upstream server and blocked in there?
How is it possible to see "query **blocked**" in query log and "**Not found in your filter lists**" in "Custom filtering rules" -> "Check the filtering"?

### Screenshots

<details><summary>Screenshot:</summary>

**1) QueryLog (AGH):**

![adh1](https://user-images.githubusercontent.com/46353488/107970807-e661dd80-6fb1-11eb-85ba-d952ce705e82.PNG)

**2) QueryLog - details (AGH):**

![adh2](https://user-images.githubusercontent.com/46353488/107970824-ec57be80-6fb1-11eb-8bb6-75c5ac6f6661.PNG)

**3) NextDNS Log**

![adh3](https://user-images.githubusercontent.com/46353488/107970833-f11c7280-6fb1-11eb-9274-0072039bbd8d.PNG)

**4) Custom Filtering Rules - check domain (AGH):**

![adh4](https://user-images.githubusercontent.com/46353488/107971446-c383f900-6fb2-11eb-97cd-9248326fc645.PNG)

</details>

**Comment by Aikatsui** [comment]
It has to check the domain for know the CNAME or IP then block and custom filtering rules component doesn't support for test it

**Comment by ameshkov** [comment]
I don't see what's causing confusion here. The query was blocked, i.e. the client that requested it received a "blocked" response.

In order to check CNAMEs and IPs AGH has to get the original response from somewhere so that we could inspect it.

**Comment by ameshkov** [comment]
> How is it possible to see "query blocked" in query log and "Not found in your filter lists" in "Custom filtering rules" -> "Check the filtering"?

Hm, well, that's a good thing, though. Worthy of a separate issue.

"Check the filtering" currently does not check the domain CNAME/IPs against the filter lists.

**Comment by ExternalDatabase** [comment]
> 
> 
> I don't see what's causing confusion here. The query was blocked, i.e. the client that requested it received a "blocked" response.
> 
> In order to check CNAMEs and IPs AGH has to get the original response from somewhere so that we could inspect it.

No it didnt recieve "blocked"... That's the point.
If I change upstream DNS to another provider i.e. Quad9 - I see "blocked" in AGH, but IP is returned to client device.

More, I see I have 140.000 queries blocked in AGH and that issue I reported only happen to few of them, not all.

And I expect that AGH will be blocking all queries that match my list and the domain "target.aws.amazon.com" is on my list - so it should be dropped, not proceeded to upstream DNS.

**Comment by ameshkov** [comment]
Ah, I guess I understand now.

1. You see, you have a blocklist with a rule that blocks `0.0.0.0` IP address.
2. AGH receives this IP because the query to `target.aws.amazon.com` is getting blocked by NextDNS (that returns `0.0.0.0` for blocked queries).
3. It blocks this IP and returns `0.0.0.0` instead of the original response which happens to be `0.0.0.0` by itself.
